src/java/io/bazel/rulesscala/jar/JarHelper.java

```diff
@@ -189,24 +196,53 @@ protected void copyEntry(String name, File file) throws IOException {
           System.err.println("adding " + file);
         }
         // Create a new entry
-        long size = isDirectory ? 0 : file.length();
-        JarEntry outEntry = new JarEntry(name);
-        long newtime = normalize ? normalizedTimestamp(name) : file.lastModified();
-        outEntry.setTime(newtime);
-        outEntry.setSize(size);
-        if (size == 0L) {
-          outEntry.setMethod(JarEntry.STORED);
-          outEntry.setCrc(0);
-          out.putNextEntry(outEntry);
-        } else {
-          outEntry.setMethod(storageMethod);
-          if (storageMethod == JarEntry.STORED) {
-            outEntry.setCrc(hashFile(file));
+        if (JarHelper.isJar(name)) {
```

Comment:
WDYT about extracting this to a private method? what this does is unpack the jar into the deployable?